addons/im_livechat/static/src/legacy/public_livechat.js

```diff
@@ -242,6 +245,45 @@ var LivechatButton = Widget.extend({
             });
         });
     },
+    _createLivechatChannel: async function() {
+        const params = {
+            channel_id: this.options.channel_id,
+            anonymous_name: this.options.default_username,
+            previous_operator_id: this._livechat.getOperatorPID()[0],
+            uuid: this._livechat.getUUID(),
+        };
+        if (this._livechat.getChatbotScriptId()) {
+            params["chatbot_script_id"] = this._livechat.getChatbotScriptId();
+        }
+        const livechatData = await session.rpc(
+            "/im_livechat/create_livechat_channel",
+            params,
+        );
+        if (!livechatData || !livechatData.operator_pid) {
+            this._livechat.setOperatorPid(undefined);
+            utils.set_cookie('im_livechat_session', '', -1); // remove cookie
+            this._chatWindow.render();
+            this._chatWindow.$('.o_composer_text_field').prop('disabled', true)
+        } else {
+            this._livechat =  new WebsiteLivechat({
+                parent: this,
+                data: livechatData,
+            });
+            this._chatWindow.updateThread(this._livechat);
+            this._updateSessionCookie();
+            this.call('bus_service', 'addChannel', this._livechat.getUUID());
+            this.call('bus_service', 'startPolling');
+        }
+    },
+    _updateSessionCookie() {
+        utils.set_cookie('im_livechat_session', utils.unaccent(JSON.stringify(this._livechat.toData()), true), 60 * 60);
+        utils.set_cookie('im_livechat_auto_popup', JSON.stringify(false), 60 * 60);
+        // livechatData.operator_pid contains a tuple (id, name)
+        // we are only interested in the id
+        const operatorPidId = this._livechat.getOperatorPID()[0];
+        const oneWeek = 7 * 24 * 60 * 60;
+        utils.set_cookie('im_livechat_previous_operator_pid', operatorPidId, oneWeek);
```

Comment:
This is code duplication with end of `_openChat`.
You could just call this function in `_openChat`, so that we clearly see it's the same code.